src/java.base/share/classes/java/net/Socket.java

```diff
@@ -497,13 +497,10 @@ private Socket(SocketAddress address, SocketAddress localAddr, boolean stream)
             if (localAddr != null)
                 bind(localAddr);
             connect(address);
-        } catch (IOException | IllegalArgumentException | SecurityException e) {
-            try {
-                close();
-            } catch (IOException ce) {
-                e.addSuppressed(ce);
-            }
-            throw e;
+        } catch (IOException | IllegalArgumentException | SecurityException error) {
```

Comment:
Naming the exception "error" is a bit strange as this isn't catching Errors. I think the only change need to the Socket constructor is to change line 500 to catch Throwable, leave the try-catch to add the suppressed exception as it's much more readable.

Reply:
Fixed in 7e633d8d213a950304c550f93a1697c1f2530a05.